x-pack/legacy/plugins/uptime/public/state/api/ping.ts

```diff
@@ -5,24 +5,66 @@
  */
 
 import { APIFn } from './types';
-import { GetPingHistogramParams, HistogramResult } from '../../../common/types';
+import {
+  PingsResponseType,
+  PingsResponse,
+  GetPingsParams,
+  GetPingHistogramParams,
+  HistogramResult,
+} from '../../../common/runtime_types';
 import { apiService } from './utils';
 import { API_URLS } from '../../../common/constants/rest_api';
 
+export const mergeParams = (
+  params: Record<string, any>,
+  maybeParams: Record<string, any>
+): Record<string, any> => {
+  let definedParams = params;
+  Object.keys(maybeParams).forEach(param => {
+    if (maybeParams[param]) {
+      definedParams = {
+        ...definedParams,
+        [param]: maybeParams[param],
+      };
+    }
+  });
+  return definedParams;
+};
```

Comment:
Is this necessary, doesn't kibana http query takes care of it?

Reply:
Right you are, deleted in https://github.com/elastic/kibana/pull/59392/commits/adf02e3e72e3e083e71d401a179cd206ad1d9ce1.